src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs

```diff
@@ -4,22 +4,34 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Linq;
 using System.Net.WebSockets;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.WebSockets.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
 using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.WebSockets
 {
     public class WebSocketMiddleware
     {
         private readonly RequestDelegate _next;
         private readonly WebSocketOptions _options;
+        private readonly ILogger _logger;
 
+        [Obsolete("This constructor has been replaced with an equivalent constructor which requires an ILoggerFactory.")]
         public WebSocketMiddleware(RequestDelegate next, IOptions<WebSocketOptions> options)
```

Comment:
There's a bug with the middleware activator where it always takes the first constructore. Move this down.

Reply:
Well that's super sketchy :(. Is this something we could fix in 3.0?

Reply:
https://github.com/aspnet/Home/issues/3091